src/Corrade/Utility/Test/MacrosCpp14Test.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+    This file is part of Corrade.
+
+    Copyright © 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016,
+                2017, 2018, 2019, 2020, 2021, 2022
+              Vladimír Vondruš <mosra@centrum.cz>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include "Corrade/TestSuite/Tester.h"
+#include "Corrade/Utility/Macros.h"
+
+namespace Corrade { namespace Utility { namespace Test { namespace {
+
+struct MacrosCpp14Test: TestSuite::Tester {
+    explicit MacrosCpp14Test();
+    void constexpr14();
+    static CORRADE_CONSTEXPR14 int sumInAStupidWay(int number);
+};
+
+CORRADE_CONSTEXPR14 int MacrosCpp14Test::sumInAStupidWay(int number) {
+    int sum = 0;
+    for(int i = 0; i != number; ++i)
+        sum += i;
+    return sum;
+}
+
+template<int> struct sfinae : std::true_type{};
+template<class T> sfinae<(T::sumInAStupidWay(0), 0)> check(int);
+template<class> std::false_type check(...);
+template<class T> struct hasRelaxedConstexpr : decltype(check<T>(0)){};
```

Comment:
But this way you're not testing if the macro is actually defined to `constexpr` when it should be, no? Like, the test would pass even if `CORRADE_CONSTEXPR14` would be empty always, or if you wouldn't use it at all.

... and you're not using the static assert anymore :P

There really needs to be something based on whether the macro is empty or not.

Reply:
Alright, so I added an `#if 0` so that it can be verified. Sadly static assert broke the build.

Reply:
I don't like this solution, to be honest. Because it requires manual fiddling with the `#if 0`, it's impossible to test on the CI, and thus it's the same as if the whole test file wouldn't be here.

Reply:
What then, you want the test to fail rather than skip?

Reply:
The test would pass even if `CORRADE_CONSTEXPR14` wasn't used anywhere in it, that's the problem I have with it. It's not testing the thing it should be testing, so it's useless.

Reply:
It'll skip on and old compiler, that's why that sfinae logic is there in the first place. Toggle the `#if 0` and see.